esp/services/ws_smc/ws_smcService.cpp

```diff
@@ -2254,3 +2256,96 @@ void CWsSMCEx::setActiveWUs(IEspContext &context, IEspActiveWorkunit& wu, IEspAc
     }
 }
 
+bool CWsSMCEx::onGetLocks(IEspContext& context, IEspGetLocksRequest& req, IEspGetLocksResponse& resp)
+{
+#ifdef _USE_OPENLDAP
+    CLdapSecManager* secmgr = dynamic_cast<CLdapSecManager*>(context.querySecManager());
+    if(secmgr && !secmgr->isSuperUser(context.queryUser()))
+        throw MakeStringException(ECLWATCH_ADMIN_ACCESS_DENIED, "Access denied, administrators only.");
+#endif
+    if (queryDaliServerVersion().compare("6.0") < 0)
+        throw MakeStringException(ECLWATCH_ADMIN_ACCESS_DENIED, "Dali server (version < 6.0) does not support WsSMC.GetLocks.");
+
+    class CSMCGetLockHelper
+    {
+    public:
+        CSMCGetLockHelper() {};
+
+        StringBuffer& readGetLockFilters(IEspGetLocksRequest& req, StringBuffer& filterBuf)
+        {
+            if (!req.getFileLockOnly_isNull() && req.getFileLockOnly())
```

Comment:
I know not new, but curious what/why the *_isNull()'s methods are for/why they are required.
Why can't if (req.getFileLockOnly()) be enough?